server/src/main/java/org/elasticsearch/indices/SystemIndexDescriptor.java

```diff
@@ -81,10 +191,160 @@ public String getDescription() {
 
     @Override
     public String toString() {
-        return "SystemIndexDescriptor[pattern=[" + indexPattern + "], description=[" + description + "]]";
+        return "SystemIndexDescriptor[pattern=[" + indexPattern + "], description=[" + description + "], aliasName=[" + aliasName + "]]";
+    }
+
+    public String getMappings() {
+        return mappings;
+    }
+
+    public Settings getSettings() {
+        return settings;
+    }
+
+    public String getAliasName() {
+        return aliasName;
+    }
+
+    public int getIndexFormat() {
+        return this.indexFormat;
+    }
+
+    public String getVersionMetaKey() {
+        return this.versionMetaKey;
+    }
+
+    public boolean isAutomaticallyManaged() {
+        return this.mappings != null || this.settings != null;
+    }
+
+    public String getOrigin() {
+        return this.origin;
     }
 
-    // TODO: Index settings and mapping
     // TODO: getThreadpool()
     // TODO: Upgrade handling (reindex script?)
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private String indexPattern;
+        private String primaryIndex;
+        private String description;
+        private XContentBuilder mappingsBuilder = null;
+        private Settings settings = null;
+        private String aliasName = null;
+        private int indexFormat = 0;
+        private String versionMetaKey = null;
+        private String origin = null;
+
```

Comment:
How about adding a private constructor so that all creations are from `SystemIndexDescriptor.builder()`?